kubeflow/core/tf-job.libsonnet

```diff
@@ -70,11 +69,49 @@
       }      
     },  // tfJobDeploy
   
-    configMap: {
+    // Default value for 
+    defaultControllerConfig(tfDefaultImage):: {
+      grpcServerFilePath: "/opt/mlkube/grpc_tensorflow_server/grpc_tensorflow_server.py",      
+    }
+    +   if tfDefaultImage != "" && tfDefaultImage != "null" then
+        {
+          tfImage: tfDefaultImage,
+        }
+        else
+        {},
+
+    azureAccelerators:: {
```

Comment:
Although it seems as if we this is missing the cuDNN library. Can you confirm @jlewi ?

Is the cuDNN installation handled by `alpha.kubernetes.io/nvidia-gpu` and the volumes are just what are required before that image can be run?

Reply:
K8s has added a mechanism "device-plugins" to install the driver. Libraries like cuDNN and CUDA can be installed in the Docker container itself.

Mounting the device drivers into the container is actually not the preferred method. Device plugin is. The reason we do this for Azure is because its not clear if AKS supports it yet.

The reason we have Azure and not other clouds is because the Azure folks added it. I'm not sure what other cloud providers would need.